EventAuxReader.h

```diff
@@ -1,24 +1,68 @@
 #if !defined(EventAuxReader_h)
 #define EventAuxReader_h
 
+#include <cassert>
+
 #include "EventIdentifier.h"
+#include "art/EventAuxiliary.h"
 #include "cms/EventAuxiliary.h"
 
+#include "TFile.h"
+class TBranch;
+
 namespace cce::tf {
-class EventAuxReader {
-public:
-  EventAuxReader(void** iAddress): address_(iAddress){}
-
-  EventIdentifier doWork() {
-    if(address_) {
-      auto aux = *reinterpret_cast<edm::EventAuxiliary**>(address_); 
-      return EventIdentifier{aux->run(), aux->luminosityBlock(), aux->event()};
-    } else {
-      return EventIdentifier{0,0,0};
+
+  inline EventIdentifier cmsEventID(void** address) {
+    assert(address);
+    auto aux = *reinterpret_cast<edm::EventAuxiliary**>(address);
+    return EventIdentifier{aux->run(), aux->luminosityBlock(), aux->event()};
+  }
+
+  inline EventIdentifier artEventID(void** address) {
+    assert(address);
+    auto aux = *reinterpret_cast<art::EventAuxiliary**>(address);
+    return EventIdentifier{aux->run(), aux->subRun(), aux->event()};
+  }
+
+  inline std::function<EventIdentifier(void**)> select_reader(TFile& file)
+  {
+    if (file.GetKey("RootFileDB")) {
+      return artEventID;
     }
+    return cmsEventID;
   }
-private:
-  void** address_;
-};
+
+  class EventAuxReader {
+  public:
+    EventAuxReader() = default;
```

Comment:
This default constructor appears to be making it easy to forget to set up the object.

Reply:
Hmm...I'll try to think of an alternative.